### PR TITLE
Display errors if a test TLS connection fails

### PR DESCRIPTION
**Pull request by mattcaswell**
If a connection fails it is useful during debugging to have any errors that are on the OpenSSL error stack at the time of the failure.

### CODE REVIEW

File: test/ssltestlib.c [hunk 1]
```diff
@@ -938,6 +938,8 @@ int create_bare_ssl_connection(SSL *serverssl, SSL *clientssl, int want,
 
         if (!clienterr && retc <= 0 && err != SSL_ERROR_WANT_READ) {
             TEST_info("SSL_connect() failed %d, %d", retc, err);
+            if (want != SSL_ERROR_SSL)
```
Comment (slontis) [inline test/ssltestlib.c]: A description of 'want' at the top might be worth doing at some point..

**Comment by mattcaswell** [comment]
I added a description of the `want` parameter. Please take another look.

**Comment by mattcaswell** [comment]
> The idea of the SSL_ERROR_SSL check is that the caller expects a failure and so we shouldn't be clearing the error queue?

We expect a failure and so we shouldn't be printing it - that will just confuse the output and make it look like an error occurred - when if fact that is expected test behaviour.

**Comment by mattcaswell** [comment]
...and yes we want to retain the reason code on the error stack.

**Comment by mattcaswell** [comment]
Pushed. Thanks.

**Pull request closed without merging**